aldryn_jobs/forms.py

```diff
@@ -133,7 +133,7 @@ def is_edit_action(self):
 
     def validate_field_uniqueness_with_app_config(self, field_name,
                                                   error_message):
-        field = self.cleaned_data[field_name]
+        field = self.cleaned_data.get(field_name, '')
```

Comment:
it matters because it makes a query that we know will return no results, so hits the db for no reason. Please make a condition that checks if this field has a value and if not then simply use found = objects.none()